nexus/src/context.rs

```diff
@@ -168,27 +168,34 @@ impl ServerContext {
             nexus_config::Database::FromUrl { url } => url.clone(),
             nexus_config::Database::FromDns => {
                 info!(log, "Accessing DB url from DNS");
-                let address = loop {
+                // It's been requested but unfortunately not supported to directly
+                // connect using SRV based lookup.
+                // TODO-robustness: the set of cockroachdb hosts we'll use will be
+                // fixed to whatever we got back from DNS at Nexus start. This means
+                // a new cockroachdb instance won't picked up until Nexus restarts.
+                let addrs = loop {
                     match resolver
-                        .lookup_socket_v6(ServiceName::Cockroach)
+                        .lookup_all_socket_v6(ServiceName::Cockroach)
```

Comment:
The last commit `[be1b1b1](https://github.com/oxidecomputer/omicron/pull/3783/commits/be1b1b14ad49649c463b138b2c5927e56fad9ee8) changed this to doing the lookup in nexus vs using the `SRV` targets and letting libpq or whoever do it at connection time. Going a little back and forth on both options. (The explicit option would make the initial nameserver/resolv.conf stuff in the nexus zone moot too).

Reply:
If we had a situation where all the CRDB servers rotated, but Nexus stayed online, this would eventually run out of the ability to contact cockroach services, right?

I have a preference for "whatever solution makes Nexus rely on hardcoded addresses" -- whether or not that's "right away" or "after everyone in the set at a particular point-in-time evacuates".

If we left this up to libpq to do name resolution, wouldn't we be able to continue accessing CRDB nodes, even if they **all* moved to new locations?

Reply:
>If we had a situation where all the CRDB servers rotated, but Nexus stayed online, this would eventually run out of the ability to contact cockroach services, right?

i think both approaches as-is are still susceptible to that. Using either addresses directly or hostnames like `{UUID}.host.control-plane.oxide.internal` that are fixed at Nexus startup will stop working if all those CRDB nodes get cycled out.

Originally was hoping to use the `_cockroach._tcp.control-plane.oxide.internal` SRV record which would work as long as internal-dns was consistent with the set of servers to use but that's not possible as-is. But I wonder if we could do this at the pool-level with a custom `bb8::ManageConnection::connect` impl that resolves the addrs there.

Reply:
Will do follow-up changes to more intelligently handle this and leave this PR as an incremental improvement by just passing all the crdb hosts instead of just one.